datacube/utils/geometry/_base.py

```diff
@@ -525,7 +525,8 @@ def __bool__(self):
         return not self.is_empty
 
     def __eq__(self, other):
-        return self.crs == other.crs and self._geom.Equal(other._geom)  # pylint: disable=protected-access
+        return (hasattr(other, 'crs') and self.crs == other.crs and
```

Comment:
So if this is just for `other is None` case why not check for that directly? Or do we expect comparisons with other compatible types?

Reply:
I don't think we are expecting comparisons with other types, but we weren't expecting comparisons with `None`, and that is what virtual products are doing.

Reply:
Another alternative is to check `isinstance(other, Geometry)`...

But we're totally nitpicking here, and I think Andrew's change is okay, if a little bit ugly, but better than an explicit `None` check.. :)

Reply:
Can `_geom` be `None`, cause this will still break in this case, what about crs being None, is crs comparison takes care of that?

If you do want to use dynamic access getattr is better as it is threadsafe and probably more performant